hpx/runtime/resource/detail/partitioner.hpp

```diff
@@ -127,6 +127,14 @@ namespace hpx { namespace resource { namespace detail
         // resource partitioner called in hpx_init
         void configure_pools();
 
+        // returns the number of threads requested by the user at start
+        // should not be called before the RP has parsed the config and
+        // assigned affinity data
+        std::size_t threads_required() {
+            HPX_ASSERT(cores_needed_ != std::size_t(-1));
+            return cores_needed_;
```

Comment:
Requested? Required? Needed? Threads? Cores? Maybe the member variable could be renamed if it's actually talking about threads and not cores.

Reply:
This is a more problem with HPX itself. HPX uses threads in many places where cores is implied and vice versa. (For example I changed all the variable names in my scheduler to refer to queue_ids, but you changed them back to thread id's when that was not what I intended). This is just more of the same. I'll give it another name.

Reply:
Agreed, so let's try to make it more consistent step by step? Required and needed are basically synonyms so I'd prefer not to have both uses. Requested is slightly different (the comment above the function uses requested) than required.

And by the way, I'm just suggesting a change here. If you had good reasons for choosing a name I'm just asking you to clarify why. The code doesn't tell me why.